daemon/containerd/mount.go

```diff
@@ -19,15 +22,21 @@ func (i *ImageService) Mount(ctx context.Context, container *container.Container
 		return err
 	}
 
-	// The temporary location will be under /var/lib/docker/... because
-	// we set the `TMPDIR`
-	root, err := os.MkdirTemp("", fmt.Sprintf("%s_rootfs-mount", container.ID))
-	if err != nil {
-		return fmt.Errorf("failed to create temp dir: %w", err)
-	}
-
-	if err := mount.All(mounts, root); err != nil {
-		return fmt.Errorf("failed to mount %s: %w", root, err)
+	root := ""
+	if container.State != nil && !container.State.Running && container.Pid == 0 {
```

Comment:
^^ actually;

- `container.Pid` is `container.State.Pid`, so currently it would panic if `container.State` is `nil`
- Given that `Running` and `Paused` are not mutually exclusive; which branch should `Paused` fall into?

Reply:
So perhaps we need an early return (error) at least if `container.State` is `nil`